-Xfatal-warnings does not work in metabuild

## steps

```bash

$ cat build.sbt
fork in Compile := true

$ cat project/build.properties
sbt.version=1.5.0

$ cat project/plugins.sbt
scalacOptions += "-Xfatal-warnings"

$ sbt compile
[info] welcome to sbt 1.5.0 (AdoptOpenJDK Java 1.8.0_282)
[info] loading settings for project tmp-build from plugins.sbt ...
[info] loading project definition from ~/tmp/project
~/tmp/build.sbt:1: warning: method in in trait ScopingSetting is deprecated (since 1.5.0): `in` is deprecated; migrate to slash syntax - https://www.scala-sbt.org/1.x/docs/Migrating-from-sbt-013x.html#slash
fork in Compile := true
     ^
[info] loading settings for project tmp from build.sbt ...
[info] set current project to tmp (in build file:/~/tmp/)
[info] Executing in batch mode. For better performance use sbt's shell
[success] Total time: 0 s, completed Apr 7, 2021 11:08:46 AM
```

## problem

I would like to be able to make any warnings in the metabuild fatal, so we can catch them in CI (e.g. the new syntax warnings during the sbt 1.5 migration). This does not seem to work however.

## expectation

Adding `-Xfatal-warnings` to `project/plugins.sbt` makes warnings in `build.sbt` fatal.

## notes

Adding other compiler options, e.g. `scalacOptions += "-Xlint"` to `project/plugins.sbt` seems to work fine (it now spits out a bunch of "unused import" warnings), so this might be specific to fatal warnings?

with the Scala 2.12.17 compiler, I can supply `-Xfatal-warnings` either before or after `-Wconf:...` and it still takes effect, as seen in the following transcript:

```
% cat O.scala
object O { 2 }
% /usr/local/scala/scala-2.12.17/bin/scalac -Xfatal-warnings -Wconf:cat=unused-nowarn:s O.scala
O.scala:1: warning: a pure expression does nothing in statement position; multiline expressions may require enclosing parentheses
object O { 2 }
           ^
error: No warnings can be incurred under -Xfatal-warnings.
one warning found
one error found
% /usr/local/scala/scala-2.12.17/bin/scalac -Wconf:cat=unused-nowarn:s -Xfatal-warnings O.scala
O.scala:1: warning: a pure expression does nothing in statement position; multiline expressions may require enclosing parentheses
object O { 2 }
           ^
error: No warnings can be incurred under -Xfatal-warnings.
one warning found
one error found
```

so it doesn't seem like this could be caused by scalac, so something must be wrong on the sbt side

but it isn't clear to me _how_ this could be going wrong on the sbt side

I had hoped that running `sbt -debug` would show me that command line sbt was using to compile the build definition, but it doesn't

--- a/sbt-app/src/sbt-test/project-load/werror/build.sbt
+++ b/sbt-app/src/sbt-test/project-load/werror/build.sbt
@@ -1,0 +1,6 @@
+scalaVersion := "3.7.4"
+
+@deprecated
+def foo = true
+
+fork := foo

--- a/sbt-app/src/sbt-test/project-load/werror/changes/build.sbt
+++ b/sbt-app/src/sbt-test/project-load/werror/changes/build.sbt
@@ -1,0 +1,6 @@
+scalaVersion := "3.7.4"
+
+@deprecated
+def foo = true
+
+fork := !foo

--- a/sbt-app/src/sbt-test/project-load/werror/changes/p.sbt
+++ b/sbt-app/src/sbt-test/project-load/werror/changes/p.sbt
@@ -1,0 +1,1 @@
+scalacOptions += "-Werror"

--- a/sbt-app/src/sbt-test/project-load/werror/test
+++ b/sbt-app/src/sbt-test/project-load/werror/test
@@ -1,0 +1,6 @@
+> name
+
+$ copy-file changes/p.sbt project/p.sbt
+$ copy-file changes/build.sbt build.sbt
+
+-> reload

